docs/hydra/guides/client-token-expiration.mdx

```diff
@@ -0,0 +1,205 @@
+  # - Linux/macOS:
+  #    export TTL_LOGIN_CONSENT_REQUEST=<value>
+  # - Windows Command Line (CMD):
+  #    > set TTL_LOGIN_CONSENT_REQUEST=<value>
+  #
+  login_consent_request: 1h
+ #....
+```
+
+### Access Token expiration
+
+The `ttl.access_token` key defines the expiration time of access tokens.
+
+```json title="path/to/hydra/config.yaml"
+#....
+  ## access_token ##
+  #
+  # Defines how long access tokens are valid.
+  #
+  # Default value: 1h
+  #
+  # Set this value using environment variables on
+  # - Linux/macOS:
+  #    export TTL_ACCESS_TOKEN=<value>
+  # - Windows Command Line (CMD):
+  #    > set TTL_ACCESS_TOKEN=<value>
+  #
+  access_token: 1h
+   #....
+```
+
+### Refresh token expiration
+
+The `ttl.refresh_token` key defines the expiration time of refresh tokens.
+
+:::tip
+
+When set to `-1`, refresh tokens never expire.
+
+:::
+
+```json title="path/to/hydra/config.yaml"
+#....
+  ## refresh_token ##
+  #
+  # Defines how long refresh tokens are valid. Set to -1 for refresh tokens to never expire.
+  #
+  # Default value: 720h
+  #
+  # Set this value using environment variables on
+  # - Linux/macOS:
+  #    export TTL_REFRESH_TOKEN=<value>
+  # - Windows Command Line (CMD):
+  #    > set TTL_REFRESH_TOKEN=<value>
+  #
+  refresh_token: "-1"
+    #....
+```
+
+### ID token expiration
+
+The `ttl.id_token` key defines the expiration time of ID tokens.
+
+```json title="path/to/hydra/config.yaml"
+#....
+  ## id_token ##
+  #
+  # Defines how long id tokens are valid.
+  #
+  # Default value: 1h
+  #
+  # Set this value using environment variables on
+  # - Linux/macOS:
+  #    export TTL_ID_TOKEN=<value>
+  # - Windows Command Line (CMD):
+  #    > set TTL_ID_TOKEN=<value>
+  #
+  id_token: 1h
+     #....
+```
+
+### Auth code expiration
+
+The `ttl.auth_code` defines the expiration time of auth codes.
+
+```json title="path/to/hydra/config.yaml"
+#....
+  ## auth_code ##
+  #
+  # Defines how long auth codes are valid.
+  #
+  # Default value: 10m
+  #
+  # Set this value using environment variables on
+  # - Linux/macOS:
+  #    export TTL_AUTH_CODE=<value>
+  # - Windows Command Line (CMD):
+  #    > set TTL_AUTH_CODE=<value>
+  #
+  auth_code: 1h
+     #....
+```
+
+## Lifespan configuration for specific clients
+
+To adjust the configuration for a specific client, call the
+[OAuth2ClientLifespans](../reference/api#operation/UpdateOAuth2ClientLifespans) endpoint of the Hydra admin API. This endpoint
+allows to configure every combination of client, grant type, and token type.
+
+:::tip
+
+The configuration applied to a specific client using this endpoint precedes the instance-wide configuration set in the Hydra
+configuration file.
+
+:::
+
+For example, to update the configuration of a client registered in a Hydra instance that runs locally, send this request:
+
+```shell
+$ curl -X PUT localhost:4445/clients/db779000-01f3-4c12-bf70-ff85933d485e/lifespans --data '{
```

Comment:
Cool idea to use a curl command to show this!

I think this is missing the `Content-Type: application/json` header :) And the URL probably needs `http://` prefix

Reply:
done